ocs/rook.py

```diff
@@ -204,3 +233,155 @@ def create_cephblockpool(
+
+        with open(self.CEPHFS_CURRENT_CONFIG, 'w') as conf:
+            yaml.dump(resource, conf, default_flow_style=False)
+
+        logger.info(f"Creating ceph FileSystem {self._name}")
+        assert self.CEPHFS.create(yaml_file=self.CEPHFS_CURRENT_CONFIG)
+        self.conf = self.get_current_conf()
+        self.active_count = self.conf.spec.metadataServer.activeCount
+
+    @property
+    def name(self):
+        return self._name
+
+    def get_current_conf(self):
+        """
+        A function to create in memory represenation
+        of fs config (Munchified data)
+
+        Returns:
+            Munchified object
+
+        # TODO: Get current config from openshift rather than temp conf file
+        """
+        with open(self.CEPHFS_CURRENT_CONFIG, 'r') as conf:
+            cur_conf = munchify(yaml.safe_load(conf))
+        return cur_conf
+
+    def set_current_conf(self, conf_obj):
+        """
+        After modification of in memory fs config we will call this function
+
+        Params:
+            modifiedconf(Munchobj): A munchified obj with new conf
+        """
+        with open(self.CEPHFS_CURRENT_CONFIG, 'w') as conf:
+            yaml.dump(conf_obj.toDict(), conf, default_flow_style=False)
+
+    def health_ok(self):
+        """
+        A all round health check of fs
+
+        Args:
+            None
+
+        Returns:
+            bool: True if healthy else False
+
+        """
+        assert self.check_mds_health()
+
+    def check_mds_health(self):
+        """
+        Asses health of mds
+
+        Returns:
+            bool: True if mds healthy
+        """
+        self.get_mds_info()
+        # TODO: need  checklist of health check parameters
+        # as of now just return True
+        return True
+
+    def get_mds_info(self):
+        """
+        Get mds info and fill in the pod object in self.mdss
+        """
+        #TODO: Replace api_client with OCP before merge
+        mdss = self._api_client.get_pods(
+            namespace=self.namespace,
+            label_selector=f'app == {self.MDS_APP_LABEL}'
+        )
+
+        for mds in mdss:
+            labels = self._api_client.get_pod_labels(
+                pod_name = mds,
+                namespace = self.namespace,
+            )
+            mds_id = labels['mds']
+            pod = Pod(mds, self.namespace, labels)
+            self.mdss.update({mds_id: pod})
+
+    def delete(self):
+        """
+        Destroy this fs
+
+        Returns:
+            bool: True if success else False
+        """
+        logger.info(f"Deleting the fs {self._name}")
+        out = self.CEPHFS.delete(resource_name=self._name)
+        if f'"{self._name}" deleted' in out:
+            # wait for mds pod termination
+            return self.POD.wait_for_resource(
+                condition='',
+                selector=self.MDS_APP_LABEL,
+                to_delete=True,
+            )
+        logger.error(f"Failed to delete cephfs {self._name}")
```

Comment:
I don't think you'll reach that point if there was a failure in deletion.
please use try and except:
```python
try:
    out = self.CEPHFS.delete(resource_name=self._name)
    if f'"{self._name}" deleted' in out:
       ...
    except CommandError as ex:
        logger.error(f"Failed to delete cephfs {self._name}. Error is {ex}")
        return False
```